pkg/kubelet/secret/secret_manager.go

```diff
@@ -201,8 +201,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 5 minutes (which is a lot, but...)
+	ttl := 5 * time.Minute
```

Comment:
> This is a user-facing change, and probably demands more discussion. Maybe we shouldn't assume that one configuration can work with all cluster sizes. Can we raise the TTL only in the 5000-node cluster and document it (until kubelet can watch secrets)?

I was going to suggest something similar. So my proposal would be:
- let's introduce a flag in kubelet that will be setting this TTL
- we will leave the default at the level of 1 minute
- we will bump it for cluster with size >2000 nodes; 
The problem with that is that it will be configured on cluster startup (I guess), thus it will not work if someone will scale the cluster from 1 to 5000. But maybe that's not that big deal.